### DISCUSSION TITLE
[mini.indentscope] text object in visual selection

### DISCUSSION THREAD

**Discussion by pkazmier** [nvim-mini/mini.nvim#1034] [Q&A]
As I learn more and more vim, I've recently discovered that I'm able to enter visual selection mode, enter a text object such as `ap` to select a paragraph, and then press `ap` again to expand the selection to include another paragraph, and I can repeat that as many times as I want. What do others do if they want to expand their initial selection to the next outer indent scope?

For example, with this block of code, if I press `Vai` while my cursor is on the "if" on line 118, then lines 116 to 119 are selected as expected. But if I want to expand that selection, I, perhaps naively, expected that a second `ai` would do so. I do know, however, that while still in visual selection mode, I could type `2ai` to select the 2nd level scope, which achieves the same, but curious why another bare `ai` would not do so as well?

```lua
╭────────────────────────────────────────────────────────────────────────────────╮
│ init.lua                                                                       │
│────────────────────────────────────────────────────────────────────────────────│
│ 115 local build_codesnap = function(args)                                      │
│ 116   vim.system({ "make", "-C", args.path }, { text = true }, function(r)     │
│ 117     if r.stdout ~= "" then vim.notify(r.stdout, vim.log.levels.INFO) end   │
│ 118     if r.stderr ~= "" then vim.notify(r.stderr, vim.log.levels.ERROR) end  │
│ 119   end)                                                                     │
│ 120 end                                                                        │
╰────────────────────────────────────────────────────────────────────────────────╯
```

**Comment by echasnovski** [comment]
> For example, with this block of code, if I press `Vai` while my cursor is on the "if" on line 118, then lines 116 to 119 are selected as expected. But if I want to expand that selection, I, perhaps naively, expected that a second `ai` would do so. I do know, however, that while still in visual selection mode, I could type `2ai` to select the 2nd level scope, which achieves the same, but curious why another bare `ai` would not do so as well?

Hmm... It does for me with plain `require('mini.indentscope').setup()`:

https://github.com/echasnovski/mini.nvim/assets/24854248/b4d48998-b57a-4be0-84bd-1379ac464c0b

**Reply by pkazmier** [reply]
Hmm .. good to know. Let me go find the offender! Thanks!

**Reply by pkazmier** [reply]
Ah, it's because I had `try_as_border = true`. When I set it back to `false`, it works as I had expected. Not sure why I enabled that in the first place, but I've nuked it for now as expanding the scope is much more important to me. Thanks for confirming it was supposed to work.
